### ISSUE TITLE
DataGrid editCellTemplate error

### ISSUE THREAD

**Issue by genusP** [DevExpress/devextreme-angular#542]
devexteme version: 17.1.5
    devextreme-angular version: 17.1.5
[ x ] bug Report 
If define column wit editCellTemplate and validation rule the raise error:
> Cannot read property 'jQuery321047152335852903352' of undefined

http://plnkr.co/edit/nOQy3PWVDiHihIDS0bef?p=preview
[ ] feature request 
[ ] support inquiry

**Comment by GoshaFighten** [comment]
Assign the `dxTemplate` directive to dxTextBox directly to resolve the issue.

``` HTML
<dx-text-box [value]="data.taskId"  *dxTemplate="let data of 'editor'"></dx-text-box>
```

See the updated [plunk](http://plnkr.co/edit/pHBTwDV3zKgccPbF1Bbi?p=preview).

**Comment by genusP** [comment]
Editor not bind to field. If click add and set all values, always this column show message is require. If click edit, editor is empty. [plunk](http://plnkr.co/edit/FuKNmf2f1u4n8t9ApW1a?p=preview)

**Comment by GoshaFighten** [comment]
You need to use the [setValue](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#editCellTemplate) method in the dxTextBox onValueChanged to set a changed value back to the grid.

``` HTML
<dx-text-box [value]="data.value" (onValueChanged)="onValueChanged($event, data)"  *dxTemplate="let data of 'editor'"></dx-text-box>
```

``` TypeScript
onValueChanged(e: any, cellInfo: any) {
  cellInfo.setValue(e.value);
}
```

See the updated [plunk](http://plnkr.co/edit/yqC1FOQWRNaqNCtkaCxx?p=preview).

**Comment by GoshaFighten** [comment]
Fixed in devextreme of version 17.1.8 and 17.2.3. See the updated [plunk](http://plnkr.co/edit/U1bId6D1Te3JztN6359Z?p=preview).
